supermarq-benchmarks/supermarq/qcvv/xeb.py

```diff
@@ -0,0 +1,390 @@
+# Copyright 2021 The Cirq Developers
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tooling for cross entropy benchmark experiments.
+"""
+from __future__ import annotations
+
+import itertools
+import random
+from collections.abc import Iterable, Sequence
+from dataclasses import dataclass, field
+
+import cirq
+import numpy as np
+import pandas as pd
+import seaborn as sns
+from scipy.stats import linregress
+from tqdm.contrib.itertools import product
+from tqdm.notebook import tqdm
```

Comment:
Style nit: for external packages we tend to prefer direct references (e.g., `import scipy` + `scipy.stats.linregress(...)`) (with some exceptions, see [style guide](https://superstaq.readthedocs.io/en/latest/resources/developer_guide.html#imports))